src/components/fx/hover.js

```diff
@@ -1095,7 +1096,13 @@ function cleanPoint(d, hovermode) {
         d.yVal = d.ya.c2d(d.yLabelVal);
     }
 
-    if(d.zLabelVal !== undefined) d.zLabel = String(d.zLabelVal);
+    if(d.zLabelVal !== undefined) {
+        if(d.zformat !== undefined) {
+            d.zLabel = d3.format(d.zformat)(d.zLabelVal).replace(/-/g, constants.MINUS_SIGN);
```

Comment:
Please reuse [`Axes.tickText`](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/axes.js#L1167)

Reply:
Referencing https://github.com/plotly/plotly.js/pull/1965, maybe hover-only `format` attributes could support the notion of _stops_ as well.

Reply:
The problem with using `Axes.tickText` is that heatmaps don't have an appropriate z-axis that could be passed to `Axes.tickText`. Should I create a dummy z-axis and use that, similar to what is being done [here](https://github.com/plotly/plotly.js/blob/f1faa2871dbf182847579166af47d1cc2a983707/src/plots/cartesian/axes.js#L601)?

Reply:
> Should I create a dummy z-axis and use that, similar to what is being done here?

Yes, that's the approach we've used in several places. Actually I think we should do that anyway, even without a custom format, as basing the formatting on the actual range of `z` values in the data (ie formatting the hover text the default way we would on an `x` or `y` axis of the same range) would probably solve this issue in most cases anyway, without asking users to pick their own format.